home-manager/install.nix

```diff
@@ -31,6 +33,29 @@ runCommand "home-manager-install" {
       home.stateVersion = "19.09";
     }
     EOF
+      else
+        cat > $confFile <<EOF
+    { config, pkgs, ... }:
+
+    {
+      # Let Home Manager install and manage itself.
+      programs.home-manager.enable = true;
+
+      # Enable special tweaks for non-NixOS systems.
+      home.targetOperatingSystem = "linux";
```

Comment:
I fear some care have to be taken here for macos users 😖

Reply:
This isn't necessary for this (for now, just `linux` and `nixos` will suffice), but something that might be useful for the future, if we start adding more OS-specific workarounds:

Something that we just discussed on IRC would be something like having a function, `isLinux` that checks the value of `home.targetOperatingSystem` and returns true for `"linux"` and whatever other OS-specific workarounds we add. This way, we wouldn't need to add additional `targetOperatingSystem == "linux" || targetOperatingSystem == "centos"` every time we find a workaround that should be added for one OS, but not globally for all Linux-based distributions.

```nix
isLinux = target:
      if target == "linux"
         || target == "centos"
         || .....
      then true
      else false;
```

Reply:
This seems to be a good idea, I will add it